src/com/palmergames/bukkit/towny/TownySettings.java

```diff
@@ -1911,26 +1918,34 @@ public static boolean isUpkeepPayingPlots() {
 
 		return getBoolean(ConfigNodes.ECO_UPKEEP_PLOTPAYMENTS);
 	}
-	
-    public static double getTownPenaltyUpkeepCost(Town town) {
-    	
-    	int claimed, allowedClaims, overClaimed;
-    	
-	    if (getUpkeepPenalty() > 0) {
-	    	
-	    	claimed = town.getTownBlocks().size();
-	    	allowedClaims = getMaxTownBlocks(town);
-	    	overClaimed = claimed - allowedClaims;
-
-	    	if (!town.isOverClaimed())
-	    		return 0;
-	    	
-	    	if (isUpkeepPenaltyByPlot())
-	    		return getUpkeepPenalty() * overClaimed;
-	    	else
-	    		return getUpkeepPenalty();
-	    } else return 0;
-    }
+
+	public static double getTownPenaltyUpkeepCost(Town town) {
+		TownUpkeepPenalityCalculationEvent event = new TownUpkeepPenalityCalculationEvent(town, getTownPenaltyUpkeepCostRaw(town));
+		Bukkit.getPluginManager().callEvent(event);
+		return event.getUpkeep();
+	}
+
+	private static double getTownPenaltyUpkeepCostRaw(Town town) {
+
+		int claimed;
+		int allowedClaims;
+		int overClaimed;
+
+		if (getUpkeepPenalty() > 0) {
+
+			claimed = town.getTownBlocks().size();
+			allowedClaims = getMaxTownBlocks(town);
+			overClaimed = claimed - allowedClaims;
+
+			if (!town.isOverClaimed())
+				return 0;
+
+			if (isUpkeepPenaltyByPlot())
+				return getUpkeepPenalty() * overClaimed;
+			else
+				return getUpkeepPenalty();
+		} else return 0;
```

Comment:
Since that else is pretty much the last line, you can just remove it, and put `return 0`.